### PR TITLE
Stand up SDL on Windows platform

### PR DESCRIPTION
**Pull request by anosach-luxoft**
Fix small issues to successfully run SDL on windows native platform:
- Explicitely specify std::string type in std::make_pair to avoid cast to const char*
- Move IsRelativePath() and MakeAbsolutePath() functions to file_system to be consistent
- Fix shared pointer defect with InterlockedDecrement() returned value

### CODE REVIEW

**Comment by anosach-luxoft** [comment]
@AGaliuzov, @pestOO, @malirod, please review.

File: src/components/utils/src/file_system_win.cc [hunk 1]
```diff
@@ -395,4 +393,18 @@ bool file_system::MoveFile(const std::string& src,
   return true;
 }
 
+bool file_system::IsRelativePath(const std::string& path) {
+  if (path.empty()) {
+    return false;
+  }
+  return path.size() < 2 || ':' != path[1];
```
Comment (pestOO) [inline src/components/utils/src/file_system_win.cc]: `BOOL PathIsRelative(
  _In_ LPCTSTR lpszPath
);`?

File: src/components/utils/src/file_system_win.cc [hunk 2]
```diff
@@ -395,4 +393,18 @@ bool file_system::MoveFile(const std::string& src,
...
+}
+
+void file_system::MakeAbsolutePath(std::string& path) {
+  if (!IsRelativePath(path)) {
+    return;
+  }
+  path = file_system::CurrentWorkingDirectory() + "/" + path;
```
Comment (pestOO) [inline src/components/utils/src/file_system_win.cc]: `DWORD WINAPI GetFullPathName(
  _In_  LPCTSTR lpFileName,
  _In_  DWORD   nBufferLength,
  _Out_ LPTSTR  lpBuffer,
  _Out_ LPTSTR  *lpFilePart
);`?

**Comment by pestOO** [comment]
Reviewed

File: src/components/utils/src/file_system_win.cc (hunk 1 above)
Comment (AGaliuzov) [inline src/components/utils/src/file_system_win.cc]: What for?

**Pull request merged**